juniper/src/schema/schema.rs

```diff
@@ -295,15 +295,22 @@ impl<'a, S: ScalarValue + 'a> TypeType<'a, S> {
                         {
                             interface_names
                                 .iter()
-                                .any(|name| name == iface_name)
-                                .then(|| context.type_by_name(name))
-                                .flatten()
+                                .any(|iname| iname == iface_name)
+                                .then(|| name.as_ref())
                         } else {
                             None
                         }
                     })
-                    .collect(),
-            ),
+                    .collect();
+                type_names.sort();
```

Comment:
This just sorts a `Vec<&str>` of type names, since they are all Object types there is no need to care about fundamental type order.